src/raftstore/store/read_queue.rs

```diff
@@ -62,17 +62,48 @@ impl Drop for ReadIndexRequest {
     }
 }
 
-#[derive(Default)]
 pub struct ReadIndexQueue {
     reads: VecDeque<ReadIndexRequest>,
     ready_cnt: usize,
     // How many requests are handled.
     handled_cnt: usize,
     // map[uuid] -> offset in `reads`.
     contexts: HashMap<Uuid, usize>,
+
+    retry_timeout: usize,
+    retry_elapsed: usize,
+    last_retried: usize,
 }
 
 impl ReadIndexQueue {
+    pub fn new(cfg: &Config) -> Self {
+        ReadIndexQueue {
+            reads: VecDeque::default(),
+            ready_cnt: 0,
+            handled_cnt: 0,
+            contexts: HashMap::default(),
+            retry_timeout: cfg.raft_election_timeout_ticks,
+            retry_elapsed: 0,
+            last_retried: 0,
+        }
+    }
+
+    pub fn check_needs_retry(&mut self) -> bool {
```

Comment:
Why not pass configuration here instead? So that only one field is necessary.

Reply:
So `push_back` will also needs the configuration, which makes it more complex.